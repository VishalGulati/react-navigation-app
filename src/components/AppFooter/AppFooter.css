footer {
    background-color: #5FC3E4;
    padding: 10px;
    margin-top: 30px;
    width: 100%;
    position: fixed;
    bottom: 0;
}

a#footer-link {
    color: orangered;
}