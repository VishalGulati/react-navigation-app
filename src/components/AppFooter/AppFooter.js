
import React from 'react';
import './AppFooter.css';

const AppFooter = () => {
    return (
        <footer>
            Designed and Coded by Vishal Gulati - &copy; 2019
        </footer>
    );
}

export default AppFooter;