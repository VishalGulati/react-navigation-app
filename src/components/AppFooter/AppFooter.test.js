import React from 'react';
import ReactDOM from 'react-dom';
import AppFooter from './AppFooter';
import { shallow } from 'enzyme';

const wrapper = shallow(<AppFooter />);

it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<AppFooter />, div);
  ReactDOM.unmountComponentAtNode(div);
});

it('Component has footer element', () => {
  expect(wrapper.find('footer').length).toEqual(1);
});

it('snapshot testing', () => {
  expect(wrapper).toMatchSnapshot();
});
