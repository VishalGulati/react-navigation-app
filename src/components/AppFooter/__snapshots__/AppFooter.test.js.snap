// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot testing 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <AppFooter />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": "Designed and Coded by Vishal Gulati - © 2019",
    },
    "ref": null,
    "rendered": "Designed and Coded by Vishal Gulati - © 2019",
    "type": "footer",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": "Designed and Coded by Vishal Gulati - © 2019",
      },
      "ref": null,
      "rendered": "Designed and Coded by Vishal Gulati - © 2019",
      "type": "footer",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
